### ISSUE TITLE
SassError: Can't find stylesheet to import when it contains @forward with _index file. [linux docker container]

### ISSUE THREAD

**Issue by kintz09** [webpack-contrib/sass-loader#822]
This bug report is identical to issue #804 except that the [recommended workaround](https://github.com/webpack-contrib/sass-loader/issues/804#issuecomment-586095020) doesn't fix the issue.

The workaround does fix the compiler error on macOS and Windows, but I have yet to get the project to compile in my Bitbucket Pipeline which uses a docker container.

- _Operating System:_ Linux 4.19.76-linuxkit x86_64 [Debian GNU/Linux 9 (stretch)]
- _Node Version:_ 12.13.1
- _NPM Version:_ 6.12.1
- _webpack Version:_ 3.4.1
- _sass-loader Version_: 8.0.2
- _sass (dart-sass)_: 1.26.3
- _Docker Image_: [node:12.13.1](https://hub.docker.com/layers/node/library/node/12.13.1/images/sha256-dab9bd5a30bf7200729768ffe4bf764bff2ea4a7e15bd48d732da24414373414)

### Expected Behavior

Expected sass-loader to resolve the module when _index.scss files are used. Expected the workaround to resolve the module when running inside the docker container.

### Actual Behavior

Locally, the project compiles as expected when the workaround is implemented.

In the docker container,` sass-loader` throws following error when a sass file that resolves to `_index.scss` with an @forward statement is imported (reproducible steps and code below):
```
./src/AppBarFail.module.scss
SassError: Can't find stylesheet to import.
  ╷
2 │ @import "@material/top-app-bar";
  │         ^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  /sass-loader-bug/src/AppBarFail.module.scss 2:9  root stylesheet
```

### Code

The project linked below showcases this bug. It is a standard CRA v3 project, non-ejected, using craco to configure `webpackImporter: false`. I took the default app, setup sass, and then added a component to recreate the error. You can look at my commits to see the exact steps taken and when the project starts to fail.
https://github.com/kintz09/sass-loader-bug

### How Do We Reproduce?

1. [Install docker](https://docs.docker.com/get-docker/)
2. Pull the docker image. `docker pull node:12.13.1`
3. Checkout repo. `git@github.com:kintz09/sass-loader-bug.git`
4. Install packages. `npm install`
5. `npm run build` to confirm it compiles locally.
6. Run project in docker container. Info on options used below can be found [here in step 3](https://confluence.atlassian.com/bitbucket/debug-your-pipelines-locally-with-docker-838273569.html)
`docker run -it --volume=/Users/<user>/<pathToRepo>/sass-loader-bug:/sass-loader-bug --workdir="/sass-loader-bug" --memory=4g --memory-swap=4g --memory-swappiness=0 --entrypoint=/bin/bash node:12.13.1`
7. Build and observe the error. `npm run build`

### Other Information

- I am working from macOS. This issue does not occur on my team's Mac or Windows computers but always fails in a node docker container.
- I have confirmed the `webpackImporter: false` option is working as expected.
- I've also tested the following docker containers with the same result: node:12.16.2, node:12.15.0, node:10.20.0, ideavate/amazonlinux-node:12-build

If you know any alternate images that work, please suggest them and I'll test. Otherwise let me know if I can provide any additional information.

**Comment by evilebottnawi** [comment]
> 5. npm run build to confirm it compiles locally

I have a problem on 5 step without docker:
```
SassError: Can't find stylesheet to import.
  ╷
2 │ @import "@material/top-app-bar";
  │         ^^^^^^^^^^^^^^^^^^^^^^^
  ╵
```

**Comment by evilebottnawi** [comment]
Researching a problem

**Comment by evilebottnawi** [comment]
Problem not in `sass-loader`, you can run `node_modules/.bin/sass src/AppBarFail.module.scss` and you will get the same error, you need setup:

```js
module.exports = {
  style: {
    sass: {
      loaderOptions: {
        // See https://github.com/webpack-contrib/sass-loader/issues/804
        webpackImporter: false,
        sassOptions: { includePaths: ['node_modules'] },
        // Prefer Dart Sass ('sass') over 'node-sass' if both are installed
        implementation: require('sass')
      }
    }
  }
}
```

**Comment by evilebottnawi** [comment]
Full duplicate of #804, continue working on the issue

**Comment by kintz09** [comment]
@evilebottnawi Thank you for your help on this issue! I can confirm your recommended change fixed my issue.

After the fact I checked my bash_profile.sh and indeed there was an environment variable for SASS_PATH. This must be the reason why this issue only occured in the docker container.

I also verified by removing the environmental variable from my bash_profile. Without the sassOptions applied the error occurred. With the sassOptions applied the error did not occur. When the environmental variable was set in bash_profile, the error didn't occur as expected.

Thanks again and stay safe!
